crates/chain/src/keychain/txout_index.rs

```diff
@@ -208,11 +222,9 @@ impl<K: Clone + Ord + Debug> Indexer for KeychainTxOutIndex<K> {
             Some((descriptor_id, index)) => {
                 // We want to reveal spks for descriptors that aren't tracked by any keychain, and
                 // so we call reveal with descriptor_id
-                if let Some((_, changeset)) = self.reveal_to_target_with_id(descriptor_id, index) {
-                    changeset
-                } else {
-                    super::ChangeSet::default()
-                }
+                let (_, changeset) = self.reveal_to_target_with_id(descriptor_id, index)
+                    .expect("descriptors are added in a monotone manner, there cannot be a descriptor id with no corresponding descriptor");
```

Comment:
I think 90% is a bit of an overstatement? We still need the `next_reveal_index` and `target_index` anyway (since we need to know which spks to reveal to reveal them). The actual logic that creates the returned spk iters is two lines. However, I would simplify this method by taking in a descriptor directly and not returning an option.